File: src/main/java/dev/jason/project/spring/vc_server/data/dto/UserDto.java
package dev.jason.project.spring.vc_server.data.dto;

import dev.jason.project.spring.vc_server.domain.model.User;

public record UserDto(String uid, String displayName, String profilePictureUrl, User.Status status) {
	
	public static UserDto fromDomainUser(User user) {
		return new UserDto(user.uid(), user.displayName(), user.profilePictureUrl(), user.status());
	}
}

File: core/src/main/java/dev/jason/app/compose/core/messaging/data/dto/UserDto.kt
package dev.jason.app.compose.core.messaging.data.dto

import dev.jason.app.compose.core.messaging.domain.model.User
import kotlinx.serialization.Serializable

@Serializable
data class UserDto(
    val uid: String,
    val displayName: String,
    val profilePictureUrl: String
)

fun UserDto.toDomain() = User(uid, displayName, profilePictureUrl)